File: durmitor/src/main/java/ql/evaluator/value/Undefined.java
package ql.evaluator.value;

import ql.ast.type.Type;
import ql.visitors.interfaces.ValueVisitor;

public class Undefined extends Value<String> {

    private final String value;
    
    public Undefined() { 
        this.value = null;
    }
    
    @Override
    public String toString() {
        return this.value;
    }

    @Override
    public String getValue() {
        return value;
    }
    
    @Override
    public Type getType() {
        return new ql.ast.type.Undefined();
    }
    
    @Override
    public Value<?> accept(ValueVisitor visitor) {
        return visitor.visit(this);
    }
}

File: Erik_Vincent/Assignment1/Assignment1/Model/QL/AST/Undefined.cs
using Assignment1.Model.QL.AST.Expression;

namespace Assignment1.Model.QL.AST
{
    public class Undefined : Value
    {
        public Undefined(Type type) : base(type) { }

        public override void Accept(IExpressionVisitor visitor) => visitor.Visit(this);
    }
}
